File: Uri/URI_Python/URI1222.py
while True:
    try:
        n, l, c = map(int, input().split()) # número de palavras, linhas por página e caracteres por linha
        palavras = input().split() # lista de palavras
    except EOFError:
        break

    num_paginas = 1
    num_linhas = 1
    num_caracteres = 0

    for palavra in palavras:
        tamanho_palavra = len(palavra)

        if num_caracteres == 0:
            # primeira palavra da linha
            num_caracteres = tamanho_palavra
        else:
            # já existe pelo menos uma palavra na linha
            num_caracteres += tamanho_palavra + 1 # considerar um espaço entre as palavras

        if num_caracteres > c:
            # a palavra atual não cabe na linha, começa uma nova linha
            num_linhas += 1
            num_caracteres = tamanho_palavra

            if num_linhas > l:
                # a nova linha não cabe na página, começa uma nova página
                num_paginas += 1
                num_linhas = 1
    print(num_paginas)

File: Uri/URI_JAVA/URI1222.java
package URI_JAVA;
import java.util.Scanner;

public class URI1222 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        while (sc.hasNext()) {
            int n = sc.nextInt(); // número de palavras
            int l = sc.nextInt(); // número máximo de linhas por página
            int c = sc.nextInt(); // número máximo de caracteres por linha
            sc.nextLine(); // consumir a quebra de linha após os inteiros

            String[] palavras = sc.nextLine().split(" ");

            int numPaginas = calcularNumPaginas(n, l, c, palavras);
            System.out.println(numPaginas);
        }

        sc.close();
    }
    private static int calcularNumPaginas(int n, int l, int c, String[] palavras) {
        int numPaginas = 1;
        int numLinhas = 1;
        int numCaracteres = 0;

        for (String palavra : palavras) {
            int tamanhoPalavra = palavra.length();

            if (numCaracteres == 0) {
                // primeira palavra da linha
                numCaracteres = tamanhoPalavra;
            } else {
                // já existe pelo menos uma palavra na linha
                numCaracteres += tamanhoPalavra + 1; // considerar um espaço entre as palavras
            }

            if (numCaracteres > c) {
                // a palavra atual não cabe na linha, começa uma nova linha
                numLinhas++;
                numCaracteres = tamanhoPalavra;

                if (numLinhas > l) {
                    // a nova linha não cabe na página, começa uma nova página
                    numPaginas++;
                    numLinhas = 1;
                }
            }
        }

        return numPaginas;
    }
}
